universal-application-tool-0.0.1/app/views/HtmlBundle.java

```diff
@@ -0,0 +1,180 @@
+  private ArrayList<String> footerStyles = new ArrayList<String>();
+  private ArrayList<Tag> headScripts = new ArrayList<Tag>();
+  private ArrayList<Tag> headerContent = new ArrayList<Tag>();
+  private ArrayList<String> headerStyles = new ArrayList<String>();
+  private ArrayList<Tag> mainContent = new ArrayList<Tag>();
+  private ArrayList<String> mainStyles = new ArrayList<String>();
+  private ArrayList<EmptyTag> metadata = new ArrayList<EmptyTag>();
+  private ArrayList<Tag> stylesheets = new ArrayList<Tag>();
+  private ArrayList<ToastMessage> toastMessages = new ArrayList<ToastMessage>();
+
+  public HtmlBundle addBodyStyles(String... styles) {
+    bodyStyles.addAll(Arrays.asList(styles));
+    return this;
+  }
+
+  public HtmlBundle addFooterContent(Tag... tags) {
+    footerContent.addAll(Arrays.asList(tags));
+    return this;
+  }
+
+  public HtmlBundle addFooterScripts(Tag... sources) {
+    footerScripts.addAll(Arrays.asList(sources));
+    return this;
+  }
+
+  public HtmlBundle addFooterStyles(String... styles) {
+    footerStyles.addAll(Arrays.asList(styles));
+    return this;
+  }
+
+  public HtmlBundle addHeadScripts(Tag... sources) {
+    headScripts.addAll(Arrays.asList(sources));
+    return this;
+  }
+
+  public HtmlBundle addHeaderContent(Tag... tags) {
+    headerContent.addAll(Arrays.asList(tags));
+    return this;
+  }
+
+  public HtmlBundle addHeaderStyles(String... styles) {
+    headerStyles.addAll(Arrays.asList(styles));
+    return this;
+  }
+
+  public HtmlBundle addMainContent(Tag... tags) {
+    mainContent.addAll(Arrays.asList(tags));
+    return this;
+  }
+
+  public HtmlBundle addMainStyles(String... styles) {
+    mainStyles.addAll(Arrays.asList(styles));
+    return this;
+  }
+
+  public HtmlBundle addStylesheets(Tag... sources) {
+    stylesheets.addAll(Arrays.asList(sources));
+    return this;
+  }
+
+  public HtmlBundle addToastMessages(ToastMessage... messages) {
+    toastMessages.addAll(Arrays.asList(messages));
+    return this;
+  }
+
+  private ContainerTag getContent() {
+    return html(renderHead(), renderBody());
+  }
+
+  public String getTitle() {
+    return pageTitle;
+  }
+
+  public HtmlBundle setTitle(String title) {
+    pageTitle = title;
+    return this;
+  }
+
+  /** The page body contains: - header - main - footer */
+  private ContainerTag renderBody() {
+    ContainerTag bodyTag = j2html.TagCreator.body(renderHeader(), renderMain(), renderFooter());
+
+    if (bodyStyles.size() > 0) {
+      bodyTag.withClasses(bodyStyles.toArray(new String[0]));
+    }
+
+    return bodyTag;
+  }
+
+  private ContainerTag renderFooter() {
+    ContainerTag footerTag = footer().with(footerContent).with(footerScripts);
+
+    if (footerStyles.size() > 0) {
+      footerTag.withClasses(footerStyles.toArray(new String[0]));
+    }
+
+    return footerTag;
+  }
+
+  /**
+   * The page head contains: - page title - page metadata - CSS styles - javascript that needs to
```

Comment:
(non-blocking): I think you can format these as actual lists like here, e.g., https://github.com/seattle-uat/civiform/blob/fa6d6e42e3b7e9466ac173bb0bc2bdd86de362e3/universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java#L58